### PR TITLE
[Documentation] Added examples to entity matching

### PR DESCRIPTION
**Pull request by torbjornopheim**
## Description
Added documentation for entity matching.

## Checklist:
- [ ] Tests added/updated.
- [ ] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [x] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [x] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

**Comment by torbjornopheim** [comment]
Don't see a quick way to solve this EOF riddle

File: pyproject.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [tool.poetry]
 name = "cognite-sdk"
 
-version = "6.5.8"
+version = "6.5.9"
```
Comment (andersfylling) [inline pyproject.toml:4]: Do we need a patch release? since it's just documentation. Or is the backporting option a bit of a pain?

**Review by andersfylling — APPROVED** [review]
LGTM

**Review by erlendvollset — COMMENTED** [review]
Let's also add links to the spec (see examples in e.g. AssetsAPI), and regenerate code examples by executing /scripts/generate_code_snippets.py, and create a PR to the service contracts repo to update them.

File: cognite/client/_api/entity_matching.py [hunk 2]
```diff
@@ -109,7 +128,11 @@ def list(
             limit (int, optional): Maximum number of items to return. Defaults to 100. Set to -1, float("inf") or None to return all items.
 
         Returns:
-            EntityMatchingModelList: List of models."""
+            EntityMatchingModelList: List of models.
+
+        Examples:
+                >>> client.entity_matching.list(limit=1, name="test")
```
Comment (andersfylling) [inline cognite/client/_api/entity_matching.py:134]: seems the errors are complaining that client and the import for client does not exist?

File: pyproject.toml (hunk 1 above)
Comment (erlendvollset) [reply pyproject.toml:4]: Don't need to bump the version

**Comment by torbjornopheim** [comment]
Added links to the ones I already found here: [3cf0c28](https://github.com/cognitedata/cognite-sdk-python/pull/1123/commits/3cf0c280982b665ad45103a8f50ee5dcc14a6084)

**Pull request merged**